src/ir/properties.h

```diff
@@ -51,8 +51,26 @@ inline bool isSymmetric(Binary* binary) {
     case XorInt64:
     case EqInt64:
     case NeInt64:
+
+    case EqFloat32:
+    case NeFloat32:
+    case EqFloat64:
+    case NeFloat64:
       return true;
 
+    case AddFloat32:
+    case MulFloat32:
+    case AddFloat64:
+    case MulFloat64: {
+      // We should be more careful with different forrm of NaNs so
+      // check if LHS is non-NaN constant. We are not interested in RHS
+      // in term of canonization which prefer constant on right side.
```

Comment:
How about using this as the comment:

"If the LHS is known to be non-NaN, the operands can commute. We don't care about the RHS because right now we only know if an expression is non-NaN if it is constant, but if the RHS is constant, then this expression is already canonicalized."

Reply:
Thanks! Done